rfcs/0000-upgradeable-lifecycle.md

```diff
@@ -31,6 +29,8 @@ metadata:
 
 ```
 
+We would also need compatibility checking to make sure that kpack can support the version of the lifecycle that is being used. Additionally, we would not want all images to be rebuilt on a lifecycle upgrade. To accomplish the compatibility checking, we can create a validating webhook that would make sure that the lifecycle in the config map was supported by kpack.
```

Comment:
Note, this would require the config map to include metadata on what platform/buildpack apis its lifecycle supports.